src/Uno.UI/UI/Xaml/Media/ImageBrush.iOS.cs

```diff
@@ -39,24 +39,29 @@ public partial class ImageBrush
 			{
 				SetImage(null, failIfNull: false);
 				oldValue?.Dispose();
+				OpenImageAsync(newValue);
+			}
+		}
+
+		private async void OpenImageAsync(ImageSource newValue)
+		{
+			CoreDispatcher.CheckThreadAccess();
 
-				_imageScheduler.Disposable = CoreDispatcher.Main.RunAsync(
-					CoreDispatcherPriority.Normal,
-					async ct =>
-					{
-						var image = await Task.Run(async () => await newValue.Open(ct));
+			var cd = new CancellationDisposable();
 
-						if (ct.IsCancellationRequested)
-						{
-							return;
-						}
-						
-						SetImage(image);
-					});
+			_imageScheduler.Disposable = cd;
+
+			var image = await Task.Run(async () => await newValue.Open(cd.Token));
```

Comment:
`async` + `await` overkill here.

Reply:
Done it in another PR since this one is already merged